src/ServiceInsight/Framework/Modules/CoreModule.cs

```diff
@@ -21,6 +21,7 @@ protected override void Load(ContainerBuilder builder)
             builder.RegisterType<ServiceControlConnectionProvider>().InstancePerLifetimeScope();
             builder.RegisterType<DefaultServiceControl>().As<IServiceControl>().InstancePerLifetimeScope();
             builder.RegisterType<CommandLineArgParser>().SingleInstance().OnActivating(e => e.Instance.Parse());
+            builder.RegisterType<WorkNotifier>().As<IWorkNotifier>().SingleInstance();
```

Comment:
> It's not about consistency it's about correct operation

Agreed, correct operation is the aim. Consistency in these two registrations is coincidental. Since `IEventAggregator` is registered as `InstancePerLifetimeScope`, `WorkNotifier` cannot be registered as `SingleInstance`. It has to be either `InstancePerLifetimeScope` or `InstancePerDependency`.

I also agree that we need to review all the container registrations as a separate exercise.